### PR TITLE
Button: Extending Button from BaseButton and removing content and icon from BaseButton types

### PR DESCRIPTION
**Pull request by khmakoto**
…n from BaseButton types.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This PR makes `Button` extend a more bare bones `BaseButton` component that has no styling applied and lacks the `content` and `icon` slots to make it more similar to the native `button` component and improve perf.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9394)

### CODE REVIEW

**Comment by khmakoto** [comment]
@micahgodbolt @betrue-final-final The screener changes come from taken out the icon from screener tests, so it'd be great if I could get approval on those.

**Comment by micahgodbolt** [comment]
Do we need all of these different tests. Seems RTL isn't necessary if we aren't doing the icon

**Comment by khmakoto** [comment]
Good catch! Let me take out the RTL tests.

File: packages/experiments/src/components/Button/BaseButton/BaseButton.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { IBaseButtonProps } from './BaseButton.types';
+import { useButtonState as state } from '../Button.state';
+import { ButtonView } from '../Button.view';
+
+export const BaseButton: React.StatelessComponent<IBaseButtonProps> = createComponent(ButtonView, {
+  displayName: 'BaseButton',
+  state,
+  styles,
+  tokens
+});
```
Comment (JasonGore) [inline packages/experiments/src/components/Button/BaseButton/BaseButton.ts]: The PR writeup (and my expectations I guess) are that `BaseButton` actually has no styling. However, I see a bunch of styling in `BaseButton.styles` that is also bound here...

File: packages/experiments/src/components/Button/BaseButton/BaseButton.styles.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+export const ButtonTokens: IBaseButtonComponent['tokens'] = {
+  contentPadding: '1px 6px',
+  textSize: 13.333
+};
+
+const GlobalClassNames = {
+  msButton: 'ms-Button'
+};
+
+export const ButtonStyles: IBaseButtonComponent['styles'] = (props, theme, tokens): IBaseButtonStylesReturnType => {
```
Comment (JasonGore) [inline packages/experiments/src/components/Button/BaseButton/BaseButton.styles.ts]: Should this file be `Button.styles` instead?

**Comment by aneeshack4** [comment]
Just putting the do not merge label on for now so no PRs get merged while we're locked down finishing up fabric 7.

**Pull request closed without merging**